specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json

```diff
@@ -1368,6 +1368,379 @@
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseRidParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The SQL database property was retrieved successfully.",
+            "schema": {
+              "$ref": "#/definitions/SqlDatabaseResource"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "DatabaseAccounts_UpdateSqlDatabase",
+        "x-ms-examples": {
+          "CosmosDBSqlDatabaseReplace": { "$ref": "./examples/CosmosDBSqlDatabaseReplace.json" }
+        },
+        "description": "Updates an existing Azure Cosmos DB SQL database",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseRidParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "updateSqlDatabaseParameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SqlDatabaseCreateUpdateParameters"
+            },
+            "description": "The parameters to provide for the current SQL database."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The SQL database update operation was successful.",
+            "schema": {
+              "$ref": "#/definitions/SqlDatabaseResource"
+            }
+          }
+        }
+      },
+      "delete": {
+        "operationId": "DatabaseAccounts_DeleteSqlDatabase",
+        "x-ms-examples": {
+          "CosmosDBSqlDatabaseDelete": { "$ref": "./examples/CosmosDBSqlDatabaseDelete.json" }
+        },
+        "description": "Deletes an existing Azure Cosmos DB SQL database.",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseRidParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The SQL database delete operation was successful."
+          }
```

Comment:
in general delete should declare 204 as a standard response indicating the requested resource does not exists/already deleted.

```
"204": {
    "description": "No Content"
}
```

Reply:
Hi, the 404 indicates content not found. Let me update 200->204 to indicate delete is successful:
https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete

Reply:
@bizarreWizard you check the service behavior here? I checked an existing delete operation [DatabaseAccounts_Delete](https://github.com/Azure/azure-rest-api-specs/blob/201bec8eac97511d69f36557c65ff7fb7aa9100c/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json#L167) and response section looks as expected for LRO delete.

```json
 "responses": {
    "202": {
        "description": "The database account delete operation will complete asynchronously."
    },
    "204": {
        "description": "The specified account does not exist in the subscription."
    }
 }
```

I was expecting something like this for SQL delete operation as well.  Please double check, how the service behaves for SQL delete LRO.

Reply:
Hi, I double run test of database and container level today,
Firstly, the Delete return "202 Accepted" by our service.
Later, the Delete status returned by our back-end is "200 OK", however, we can covert it to "204 No content".
If I delete it again, the 202 returned firstly then back-end returns 404 Not Found.